BUG: GrailsGradlePlugin CRASHES if grails-app folder does not exist.

grails-app folder should be optional.

```
 ./gradlew clean bootRun --stacktrace
```

```
Caused by: org.gradle.api.internal.plugins.PluginApplicationException: Failed to apply plugin 'org.grails.grails-web'.
        at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:173)
        at org.gradle.api.internal.plugins.DefaultPluginManager.apply(DefaultPluginManager.java:146)
        at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.lambda$applyLegacyPlugin$2(DefaultPluginRequestApplicator.java:160)
        at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.applyPlugin(DefaultPluginRequestApplicator.java:201)
        ... 164 more
Caused by: org.gradle.internal.operations.BuildOperationInvocationException: /Users/sheiberg/Documents/workspace/sitemesh3/sitemesh-examples-gsp/grails-app
        at org.gradle.internal.operations.DefaultBuildOperationRunner.throwAsBuildOperationInvocationException(DefaultBuildOperationRunner.java:192)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.access$100(DefaultBuildOperationRunner.java:24)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:75)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
        at org.gradle.api.internal.plugins.DefaultPluginManager.lambda$doApply$0(DefaultPluginManager.java:167)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:44)
        at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:166)
        ... 167 more
Caused by: java.io.FileNotFoundException: /Users/sheiberg/Documents/workspace/sitemesh3/sitemesh-examples-gsp/grails-app
        at org.grails.gradle.plugin.core.GrailsGradlePlugin.resolveGrailsSourceDirs(GrailsGradlePlugin.groovy:358)
```

workaround is to have an empty `grails-app` folder, but that is kind of silly.

--- a/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy
@@ -354,9 +354,12 @@ class GrailsGradlePlugin extends GroovyPlugin {
     @CompileStatic
     protected List<File> resolveGrailsSourceDirs(Project project) {
         List<File> grailsSourceDirs = []
-        project.file("grails-app").eachDir { File subdir ->
-            if (isGrailsSourceDirectory(subdir)) {
-                grailsSourceDirs.add(subdir)
+        File grailsApp = project.file("grails-app")
+        if (grailsApp.exists()) {
+            grailsApp.eachDir { File subdir ->
+                if (isGrailsSourceDirectory(subdir)) {
+                    grailsSourceDirs.add(subdir)
+                }
             }
         }
         grailsSourceDirs.add(project.file("src/main/groovy"))

